core/interpreter_js/src/core_to_map_bindings/mod.rs

```diff
@@ -136,7 +137,7 @@ impl<'a> JSValueDebug<'a> {
                     continue;
                 }
 
-                map.entry(&Self::Ref(key), &Self::Ref(value));
+                map.entry(&JSValueDebug::Ref(&key), &JSValueDebug::Ref(&value));
```

Comment:
rust noob question: Why here is used JSValueDebug and other places Self?

Reply:
It's related to the change that `JSValueDebug::Ref` now stores a reference to `JSValueRef` (confusing, I know). Since `JSValueDebug::Ref` is now contains a reference bound by lifetime `'a` in this context `Self` refers to a different lifetime. Considering the whole function:

```rs
impl<'a> JSValueDebug<'a> {
    fn fmt_ref(rf: &JSValueRef<'a>, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        // snip
        while let (Ok(Some(key)), Ok(value)) = (properties.next_key(), properties.next_value()) {
                // Self is really JSValueDebug<'a> (from the generic impl)
                // but we are borrowing key and value from the stack here, which is a smaller lifetime (let's call it 'b)
                // so we are really passing in the type JSValueDebug<'b> and this lifetime only lasts until the end of one while loop iteration
                // before it was okay to pass in a JSValueDebug<'a> because we weren't actually using the 'a for the ::Ref variant
                map.entry(&JSValueDebug::Ref(&key), &JSValueDebug::Ref(&value));
        } // 'b ends here
    }
}
```